plugins/core-plugin/src/main/resources/Dockerfile-template-yaml

```diff
@@ -1,16 +1,67 @@
-FROM ${baseContainerImage}
+#===================================================================#
+# Create build environment from base Python template launcher image #
+#===================================================================#
+FROM ${basePythonContainerImage} as python-base
 
-# Copy template files to /template
+# Build args
 ARG WORKDIR=/template
-COPY main.py /template
-COPY requirements.txt /template
-COPY ${yamlTemplateName} /template/template.yaml
+ARG REQUIREMENTS_FILE=requirements.txt
+ARG BEAM_VERSION=${beamVersion}
+ARG BEAM_PACKAGE=apache-beam[dataframe,gcp,test,yaml]==$BEAM_VERSION
+ARG PY_VERSION=${pythonVersion}
+
+# Copy template files to /template
+RUN mkdir -p $WORKDIR
+COPY main.py* requirements.txt* ${otherFiles} /template/
```

Comment:
Similarly, why do we need `main.py*` instead of just `main.py` here?

Reply:
Are you just not sure if these files exist? If so, I guess its fine to have here, but I still wouldn't append it to otherFiles (which presumably all should exist)

Reply:
Also, I think we are assuming main.py exists at least

Reply:
Same explanation here - you're right, no reason to wildcard main.py